python/ray/tests/test_joblib.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+import joblib
+from sklearn.datasets import load_digits
+from sklearn.model_selection import RandomizedSearchCV
+from sklearn.svm import SVC
+from ray.experimental.joblib import register_ray
+import ray
+import os
+import subprocess
+from time import time
+from joblib import Memory
+from sklearn.datasets import fetch_openml
+from sklearn.datasets import get_data_home
+from sklearn.ensemble import ExtraTreesClassifier
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.kernel_approximation import Nystroem
+from sklearn.kernel_approximation import RBFSampler
+from sklearn.pipeline import make_pipeline
+from sklearn.svm import LinearSVC
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.utils import check_array
+from sklearn.linear_model import LogisticRegression
+from sklearn.neural_network import MLPClassifier
+
+
+def test_register_ray():
+    register_ray()
+    assert "ray" in joblib.parallel.BACKENDS
+
+
+def test_ray_backend():
+    register_ray()
+    from ray.experimental.joblib._raybackend import RayBackend
+    with joblib.parallel_backend("ray"):
+        assert type(joblib.parallel.get_active_backend()[0]) == RayBackend
+
+
+def test_svm_crossvalidation():
+    digits = load_digits()
+
+    param_space = {
+        "C": np.logspace(-6, 6, 30),
+        "gamma": np.logspace(-8, 8, 30),
+        "tol": np.logspace(-4, -1, 30),
+        "class_weight": [None, "balanced"],
+    }
+
+    model = SVC(kernel="rbf")
+    search = RandomizedSearchCV(
+        model, param_space, cv=5, n_iter=100, verbose=10)
+    register_ray()
+
+    def test_local():
+        if ray.is_initialized():
+            ray.shutdown()
+        with joblib.parallel_backend("ray"):
+            search.fit(digits.data, digits.target)
+        assert ray.is_initialized()
+
+    def test_multiple_nodes():
+        if ray.is_initialized():
+            ray.shutdown()
+        subprocess.check_output(
+            ["ray", "start", "--head", "--num-cpus={}".format(4)])
```

Comment:
You should use something like the Ray cluster testing fixtures, though I'm not 100% sure what are the right instructions for setting this up (ask Edward)

Reply:
have a look at `tests/test_multi_node_2.py` for an example

Reply:
Fixed.